Validate seqtools@0.1.0

edsu7 opened the pull request:
`tests/test-job-bam.json`
- Pass validation, tests one file

`tests/test-job-fq.json`
- Fail validation. Tests multiple input files

edsu7 committed:
[wfpm v0.8.0] added starting template for validate-seqtools@0.1.0

edsu7 committed:
added `main.py`,`main.nf`, and associated test files

lindaxiang commented on validate-seqtools/main.nf:
```diff
@@ -0,0 +1,85 @@
+
+// tool specific parmas go here, add / change as needed
+params.json_file = ""
+params.skip_md5sum_check = 'NO_FILE'
```
Is boolean type more proper for `skip_md5sum_check`? If we define it as string, any string will trigger skip? like `No skipping`?

edsu7 replied:
Switched to boolean.
https://github.com/icgc-argo/argo-data-submission/blob/2ed0573fbe4a1be55ec51b67a3e56d2359b139c6/validate-seqtools/main.nf#L48

https://github.com/icgc-argo/argo-data-submission/blob/2ed0573fbe4a1be55ec51b67a3e56d2359b139c6/validate-seqtools/main.nf#L68

lindaxiang commented on validate-seqtools/Dockerfile:
```diff
@@ -0,0 +1,57 @@
+
+LABEL org.opencontainers.image.source https://github.com/icgc-argo/argo-data-submission
+LABEL org.opencontainers.image.authors Edmund Su (edmund.su@oicr.on.ca)
+LABEL org.opencontainers.image.title ICGC ARGO Data download image
```
ICGC ARGO Data Submission Validation image?

edsu7 replied:
Edited to `ICGC ARGO Data Submission Validation image`
https://github.com/icgc-argo/argo-data-submission/blob/2ed0573fbe4a1be55ec51b67a3e56d2359b139c6/validate-seqtools/Dockerfile#L5

lindaxiang commented on validate-seqtools/Dockerfile:
```diff
@@ -0,0 +1,57 @@
+ENV LANG en_US.UTF-8  
+ENV LANGUAGE en_US:en  
+ENV LC_ALL en_US.UTF-8  
+
```
What are these settings for?

edsu7 replied:
Dependency for `locales`. Added `locales` as apart of python checking for time.

lindaxiang reviewed (changes requested):
The `logs` folder and two files under expected folder need to be removed/ignored from git.

lindaxiang commented:
@edsu7 can you also add some brief description into README? :-)

lindaxiang commented on validate-seqtools/Dockerfile:
```diff
@@ -0,0 +1,57 @@
+RUN git clone https://github.com/icgc-argo/seq-tools.git &&\
+    cd seq-tools &&\
+    pip3 install -r requirements.txt  &&\
+    pip3 install .
```
I just created a new release(1.2.0) for seq-tools. Let's install stable and released seq-tools instead of the main branch.

edsu7 replied:
Added install for `1.2.1`
https://github.com/icgc-argo/argo-data-submission/blob/8ccda883ede872f40a217c87f067373b622abc5c/validate-seqtools/Dockerfile#L40

edsu7 committed:
Various updates including : `README.md` and `Dockerfile`

- removed `tests/logs`
- updated `README.md`
- Updated `Dockerfile` seq-tools install version to `1.2.1`
- switched `params.skip_md5sum_check` to boolean

edsu7 committed:
Updated Checker.nf

edsu7 commented:
> The `logs` folder and two files under expected folder need to be removed/ignored from git.

Removed `logs`:
https://github.com/icgc-argo/argo-data-submission/pull/12/commits/ba783114b019d9ac1153c9c220f74d58a272773e#diff-8fcefc7e38e7cc942a8c5cdfd62cbb31c173e8bfe4e9e2447104ce32d7ddc8fcR40

edsu7 commented:
https://github.com/icgc-argo/argo-data-submission/pull/12#issuecomment-1211285199

Added `README.md`
https://github.com/icgc-argo/argo-data-submission/blob/validate-seqtools%400.1.0/validate-seqtools/README.md

edsu7 committed:
Update Dockerfile
